pkg/driver/node.go

```diff
@@ -521,7 +525,13 @@ func (d *nodeService) nodePublishVolumeForBlock(req *csi.NodePublishVolumeReques
 	if !exists {
 		return status.Error(codes.InvalidArgument, "Device path not provided")
 	}
-	source, err := d.findDevicePath(devicePath, volumeID)
+
+	partition := ""
```

Comment:
partition would not make sense in block context, like block that has been partitioned is not block. we should throw an error imo if we find the attribute

Reply:
nvm, i think you can still partition a device wthout formatting any of the partitions right ? Not an expert on this obviously

Reply:
just checked in-tree and the code is same except in-tree never bothered to set partition of nvme volumes : / with "p#" suffix

Reply:
in-tree is going to have nvme fix in this PR https://github.com/kubernetes/kubernetes/pull/100500
I'm testing the block scenario and will update here later.

Reply:
@wongma7 just to confirm we can still partition a device without formatting.